lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -176,11 +176,19 @@ impl ShardOperation for LocalShard {
     async fn count(
         &self,
         request: Arc<CountRequestInternal>,
-        _timeout: Option<Duration>,
+        search_runtime_handle: &Handle,
+        timeout: Option<Duration>,
     ) -> CollectionResult<CountResult> {
-        // TODO run in spawn_blocking + timeout + cancellation
         let total_count = if request.exact {
-            let all_points = self.read_filtered(request.filter.as_ref())?;
+            let timeout = timeout.unwrap_or(self.shared_storage_config.search_timeout);
+            let all_points = tokio::time::timeout(
+                timeout,
+                self.read_filtered(request.filter.as_ref(), search_runtime_handle),
+            )
+            .await
+            .map_err(|_: Elapsed| {
```

Comment:
```suggestion
            .map_err(|_| {
```

Reply:
It seems we ascribed the `Elapsed` type on all `tokio::time::timeout` usages in the code base.
Maybe an artifact of previous type inference issues.
I will remove all of those in a single PR right after this one.

Reply:
Maybe we did that to avoid issues during refactoring in order to make sure the ignored value has this very type.

Reply:
For me it didn't work without specifying it for some reason

Reply:
It did build for me without the specification. Weird!